### PR TITLE
refactor: update lint, add lint docs, update git hooks

### PR DESCRIPTION
**Pull request by levibostian**
This PR is an extension to [this previous PR](https://github.com/customerio/customerio-android/pull/113) of updating ktlint to the latest version. That PR only fixed 1 branch (in-app) but all branches in the project are effected so this PR is to be merged into `develop` to fix all branches. 

Changes done in PR
1. Fix lint errors after ktlint has been updated
2. Add docs around linting with ktlint. I wrote these docs with my preferences on how linting is done in a project. The content of the docs is up for discussion. As our team scales, I felt having documentation is important for us to be on the same page. 
3. Updated git hooks. I have been using this new git hooks method in other projects and really like it. The best part is that on every git commit, ktlint formatter will automatically be executed and will run `git add` on all formatted files. No more having to make a git commit, run formatter, then run another commit. 

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-android/develop/docs/dev-notes/GIT-WORKFLOW.md).
- [ ] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request.
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1).
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request and it will automatically be merged.

### CODE REVIEW

**Comment by Shahroz16** [comment]
There are going to be super conflicting as the `Ktlint` changes are also done here 😢 
https://github.com/customerio/customerio-android/pull/112

**Comment by levibostian** [comment]
> There are going to be super conflicting as the Ktlint changes are also done here 😢
https://github.com/customerio/customerio-android/pull/112

I thought of that, too. I am not sure what to expect in regards to conflicts that will appear in the in-app branch as that's the only branch that https://github.com/customerio/customerio-android/pull/112 is in right now. 

But luckily, it might be as easy as performing a `git revert` on the in-app branch to remove the https://github.com/customerio/customerio-android/pull/112 commit, and then merging in `develop` branch into in-app branch after this PR gets merged.

**Comment by levibostian** [comment]
> When we are not linting the enums and have to manually disable 3 classes and will have to disable possibly more, I don't see the point of not disabling it globally.

Because we use `enum` classes a lot in this project for JSON values, I agree that disabling it globally wouldn't be harmful. I just checked and currently there are 5 enums in the project with 3 of those enums used for JSON values (meaning we need to have the cases be lowercase). So, we do use lowercase enums the majority of the time.

**Comment by Shahroz16** [comment]
>Because we use enum classes a lot in this project for JSON values, I agree that disabling it globally wouldn't be harmful. I just checked and currently there are 5 enums in the project with 3 of those enums used for JSON values (meaning we need to have the cases be lowercase). So, we do use lowercase enums the majority of the time.

We both agree, that the usage of enums with the lower case is more, and since BE accepts it, the usage will increase as well. with time, also that this rule wouldn't be harmful, so should we disable it globally instead of creating enum classes, then failing lint and disabling it for those closes every time. Might not make much time but would be annoying to keep track of.

File: lefthook.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Configuration for lefthook tool to manage git hooks 
+# https://github.com/evilmartians/lefthook
```
Comment (Shahroz16) [inline lefthook.yml:2]: 👍

File: common-test/src/main/java/io/customer/commontest/BaseTest.kt [hunk 2]
```diff
@@ -1,10 +1,10 @@
-package io.customer.common_test
+package io.customer.commontest
```
Comment (Shahroz16) [inline common-test/src/main/java/io/customer/commontest/BaseTest.kt:1]: According to Kotlin's recommendation, `_` can be repaced by either the single word `commontest` or `commonTest`. 

Maybe we should set criteria of when to use when? so there is not confusion?

**Comment by levibostian** [comment]
@Shahroz16, @mrehan27 this PR changes our git hooks behavior. This PR proposes that we switch to a new tool that I have found to be really enjoyable [lefthook](https://github.com/evilmartians/lefthook). The biggest benefit is that the tool really easily allows us to run the kotlin code formatter and `git add` formatted changes before each git commit. The previous git hooks setup that we were using required that you make another commit after each commit to be able to add the kotlin formatted changes. 

If this PR gets merged with this new tool in it, you will need to run these commands to get git hooks installed:

```
rm -rf .git/hooks
mkdir .git/hooks
brew install lefthook
lefthook install
```

I updated the docs in this repo with these instructions for new developers to use 👍.

**Comment by levibostian** [comment]
This task can be worked on after this PR gets merged https://github.com/customerio/issues/issues/7601

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**